### ISSUE TITLE
Unable To Import DocType If Parent Folder Already Exists

### ISSUE THREAD

**Issue by YodasMyDad** [umbraco/Umbraco-CMS#11772]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

v9.1.2

### Bug summary

I exported a DocType from one v9 site, it was in a folder called 'Compositions' within my Document Types tree. I tried to import it into another site where I already had a folder called 'Compositions' and it failed with the error below

### Specifics

```
System.InvalidOperationException, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e: A container with the same name already exists. at Umbraco.Cms.Infrastructure.Packaging.PackageDataInstallation.CreateContentTypeFolderStructure(IEnumerable`1 unsortedDocumentTypes, IEnumerable`1& entityContainersInstalled)
   at Umbraco.Cms.Infrastructure.Packaging.PackageDataInstallation.ImportDocumentTypes[T](IReadOnlyCollection`1 unsortedDocumentTypes, Boolean importStructure, Int32 userId, IContentTypeBaseService`1 service, IEnumerable`1& entityContainersInstalled)
   at Umbraco.Cms.Infrastructure.Packaging.PackageDataInstallation.ImportDocumentTypes(IEnumerable`1 docTypeElements, Int32 userId, IEnumerable`1& entityContainersInstalled)
   at Umbraco.Cms.Infrastructure.Packaging.PackageDataInstallation.ImportDocumentType(XElement docTypeElement, Int32 userId)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentTypeController.Import(String file)
   at lambda_method2113(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```

### Steps to reproduce

Just export a doctype that is in a folder within document types. Then try to import that doc type into another site that already has a folder with the same name.

### Expected result / actual result

It should just put the document type into the folder that already exists with the same name

---
_This item has been added to our backlog [AB#15697](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/15697)_

**Comment by nul800sebastiaan** [comment]
I can reproduce this, it seems to be because we're trying to put the doctype in the folder with the `FolderKeys` attribute. If that key doesn't exist we try to create a new folder with that name (and presumably that key). 

![image](https://user-images.githubusercontent.com/304656/147924558-2302b908-21ca-49f6-a45c-161074365591.png)

Indeed, if the folder already exists with the name in the `Folders` attribute then we should put the imported doctype in the existing folder.
Bonus info: you can not export folders like you can export doctypes, so there's no good way to get the same `FolderKey` between two sites.

**Comment by nul800sebastiaan** [comment]
Workaround: if you remove the `FolderKeys` attribute from the udt file, the import of the doctype will succeed and the doctype will be put in the correct folder.

**Comment by elit0451** [comment]
Fixed in #11885
